web/app/scripts/custom-reports/custom-reports-modal-controller.js

```diff
@@ -0,0 +1,169 @@
+            return RecordSchemaState.getFilterables(recordSchema.uuid).then(function(filters) {
+                _.each(filters, function(filter, path) {
+                    ctl.rowColAggs.push({
+                        label: path.split('#')[1],
+                        value: path.replace('#', ','),
+                        type: 'Filter'
+                    });
+                });
+            });
+        }
+
+        function closeModal() {
+            $modalInstance.close();
+        }
+
+        /**
+         * Helper for setting the correct params based on the type of aggregation
+         * @param {rowOrCol} Key prefix string, can be 'row' or 'col'
+         * @param {aggObj} An aggregation object from ctl.rowColAggs
+         * @param {params} An object of HTML parameters that is updated with the new key/val
+         */
+        function setRowColParam(rowOrCol, aggObj, params) {
+            var key = rowOrCol + '_';
+            if (aggObj.type === 'Time') {
+                key += 'period_type';
+            } else if (aggObj.type === 'Filter') {
+                key += 'choices_path';
+            } else if (aggObj.type === 'Geometry') {
+                key += 'boundary_id';
+            } else {
+                $log.error('Cannot set row/col param with type: ', aggObj.type);
+                return;
+            }
+            params[key] = aggObj.value;
+        }
+
+        function createReport() {
+            QueryBuilder.assembleParams(0, true, true).then(
+                function(params) {
+                    /* jshint camelcase: false */
+
+                    var crosstabsParams = {};
+                    setRowColParam('col', ctl.colAggSelected, crosstabsParams);
+                    setRowColParam('row', ctl.rowAggSelected, crosstabsParams);
+
+                    // Only allow specifying an aggregation boundary if no geographical
+                    // aggregation was selected for rows/cols. Similar logic is in the partial
+                    // for displaying/hiding the dropdown, but it is also performed here for
+                    // good measure (e.g. if selections are made and then later changed).
+                    if (ctl.geoAggSelected &&
+                        ctl.colAggSelected.type !== 'Geography' &&
+                        ctl.rowAggSelected.type !== 'Geography') {
+                        crosstabsParams.aggregation_boundary = ctl.geoAggSelected.value;
+                    }
+                    params = _.extend(params, crosstabsParams);
+
+                    // Ensure the page limit parameter is not set
+                    if (params.limit) {
+                        delete params.limit;
+                    }
+
+                    // The crosstabs endpoint uses `record_type_id` rather than `record_type`.
+                    // Renaming it here, though the name should probably be standardized.
+                    params.record_type_id = params.record_type;
```

Comment:
Oh, I think it's just the docstring that was saying `record_type_id` on the backend -- the param actually being used is `record_type`. Easy fix!